api/admin/controller/__init__.py

```diff
@@ -1796,13 +1796,31 @@ def _set_storage_external_integration_link(
                 purpose=purpose,
             )
 
+            if not current_integration_link:
+                return INTERNAL_SERVER_ERROR.detailed(
+                    "Could not create integration link."
+                )
+
             current_integration_link.other_integration_id = storage_integration.id
 
         return None
 
-    @classmethod
-    def _get_integration_protocols(cls, provider_apis, protocol_name_attr="__module__"):
+    def _get_protocol_class(self, registry, protocol_name, is_child=False):
+        api_class = registry.get(protocol_name)
+        if not api_class:
+            return None
+        if is_child:
+            if getattr(api_class, "child_settings_class", False):
```

Comment:
Rather that cls.CHILD_SETTINGS we now use a `def child_settings` implementation in the Overdrive API.